<h1>Applies to</h1>
  <ul>
    <li>C&#43;&#43;</li>
  </ul>
  <h1>What to Check For</h1>
  <p>Verify that deallocation functions don't throw exceptions.</p>
  <h1>Why</h1>
  <p>Deallocation functions may be called when exceptions are thrown. If a deallocation function throws an exception after being called when an exception is thrown, the program may terminate. This may result in a denial-of-service vulnerability.</p>
  <h1>How to Check</h1>
  <p>Search for deallocation functions. Verify that deallocation functions don't throw exceptions.</p>
  <h1>How to Fix</h1>
  <p>Identify deallocation functions that throw exceptions and rewrite them in such a way that they don't throw exceptions.</p>
  <h1>Additional Resources</h1>
  <ul>
    <li>For more information about not throwing exceptions from deallocation functions, please see <a href="https://www.securecoding.cert.org/confluence/display/cplusplus/VOID+ERR38-CPP.+Deallocation+functions+must+not+throw+exceptions">https://www.securecoding.cert.org/confluence/display/cplusplus/VOID+ERR38-CPP.+Deallocation+functions+must+not+throw+exceptions</a></li>
  </ul>
  <h1>Related Items</h1>
  <ul>
    <li>
      <a href="/article/441b4138-1866-442f-8abb-6d05236d6fcb">Don't Throw Exceptions from Deallocation Functions</a>
      <li>
        <a href="/article/fd06b089-88ea-43df-bc41-f111a1848116">Don't Throw Exceptions from Destructors</a>
        <li>
          <a href="/article/2e732e9d-5677-40cc-8e33-c40ceacc056b">
            <a href="/article/2e732e9d-5677-40cc-8e33-c40ceacc056b">Destructors don't throw exceptions</a>
          </a>
        </li>
      </li>
    </li>
  </ul>